src/fsharp/fsi/console.fs

```diff
@@ -17,7 +17,7 @@ module internal ConsoleOptions =
   let fixupRequired = false
 #else
   // Bug 4254 was fixed in Dev11 (Net4.5), so this flag tracks making this fix up version specific.
-  let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove
+  let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove && Type.GetType("Mono.Runtime") = null
```

Comment:
As far as I can interpret the issues referenced, the problem with vscode garbling utf8 chars is because the fixup code runs; in error. I'm trying to make it not run. I haven't tried it on my machine, because I don't at this point know how to replace what interactive that vscode uses. Also, I'm on OS X, not ubuntu, but perhaps build.sh works there as well?

In general, I'm trying to nudge things in the right direction, because not having a REPL that supports UTF8 is generally an issue for me.

So; hypothesis; this fixes it. I don't know how to test it, but perhaps this repository has a README that documents it?